### PR TITLE
fixes an r-group symmetrization problem

### PR DESCRIPTION
**Pull request by greglandrum**
Here's the issue:

```
In [8]: scaff = Chem.MolFromSmiles('Cc1ccccc1')
In [16]: ms = [Chem.MolFromSmiles(x) for x in ('c1ccccc1C','Fc1ccccc1C','c1cccc(F)c1C','Fc1cccc(F)c1C')]
In [17]: rdRGroupDecomposition.RGroupDecompose([scaff],ms,asSmiles=True)[0]
Out[17]:
[{'Core': 'Cc1c([*:1])cccc1[*:2]', 'R1': '[H][*:1]', 'R2': '[H][*:2]'},
 {'Core': 'Cc1c([*:1])cccc1[*:2]', 'R1': '[H][*:1]', 'R2': 'F[*:2]'},
 {'Core': 'Cc1c([*:1])cccc1[*:2]', 'R1': 'F[*:1]', 'R2': '[H][*:2]'},
 {'Core': 'Cc1c([*:1])cccc1[*:2]', 'R1': 'F[*:1]', 'R2': 'F[*:2]'}]

```

### CODE REVIEW

File: Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp [hunk 1]
```diff
@@ -363,14 +377,20 @@ double score(const std::vector<size_t> &permutation,
 
     for (std::map<std::string, int>::const_iterator it = matchSet.begin();
          it != matchSet.end(); ++it) {
-      
+#ifdef DEBUG
+      std::cerr << " equiv: " << it->first << " " << it->second << " "
+                << permutation.size() << std::endl;
+#endif
+
       // if the rgroup is a hydrogens, only consider if the group is all
```
Comment (bp-kelley) [inline Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp]: might as well fix the typo here.  `Is a hydrogens` probably tells you the mental state I was in when I wrote this.

File: Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp [hunk 2]
```diff
@@ -363,14 +377,20 @@ double score(const std::vector<size_t> &permutation,
...
       //  hydrogen, otherwise score based on the non hydrogens
-      if(it->first.find("[H]") != std::string::npos) {
-        if(static_cast<size_t>(it->second) == permutation.size())
+      if (onlyH[it->first]) {
+        if (static_cast<size_t>(it->second) == permutation.size())
           equivalentRGroupCount.push_back(static_cast<float>(it->second));
         else
-          equivalentRGroupCount.push_back(it->second * 1.0/permutation.size()); // massively downweight hydrogens
+          equivalentRGroupCount.push_back(
+              0.0);  // it->second * 1.0/permutation.size()); // massively
```
Comment (bp-kelley) [inline Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp]: Get rid of the comments.  That's what "git blame" is for ;)

**Comment by bp-kelley** [comment]
Surprisingly few changes.  We should note somewhere that the smilesSet and getSmiles are broken as well.

**Comment by greglandrum** [comment]
Ok, done. I went ahead and fixed the problem with the SMILES being combined incorrectly and added a REVIEW comment about the set.

File: Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp [hunk 3]
```diff
@@ -256,12 +258,12 @@ struct RGroupData {
       const {  // compute the canonical smiles for the attachments
     std::string s;
     for (const auto &it : smilesSet) {
-      s += it;
+      s += "."+it;
```
Comment (bp-kelley) [inline Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp]: I don't think this makes valid smiles.  I think you need

```
{
  if(s.size()) s += ".";
  s += it;
}
```

Comment (greglandrum) [reply Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp]: eh, details.
The previous smiles was also wrong. ;-)

Comment (greglandrum) [reply Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp]: errr, I mean: good point! I'll fix it!

**Comment by bp-kelley** [comment]
Thanks Greg!

**Pull request merged**